src/mlfe_typer.erl

```diff
@@ -980,9 +984,35 @@ retrieve_type(SM, M, T, [#mlfe_module{name=M, types=Ts}|Rem]) ->
 retrieve_type(SM, M, T, [_|Rem]) ->
     retrieve_type(SM, M, T, Rem).
 
--spec typ_module(M::mlfe_module(), Env::env()) -> {ok, mlfe_module()} |
-                                                  {error, term()}.
-typ_module(#mlfe_module{functions=Fs,
+-spec type_modules([mlfe_module()]) -> {ok, [mlfe_module()]} |
+                                       {error, term()}.
+type_modules(Mods) ->
+    {Pid, Monitor} = erlang:spawn_monitor(fun() ->
+        try type_modules(Mods, new_env(Mods), []) of
+            Res -> exit(Res)
+        catch
+            E:T ->
+                io:format("mlfe_typer:type_modules/2 crashed with ~p:~p~n"
+                          "Stacktrace: ~p~n", [E, T, erlang:get_stacktrace()]),
+                exit({error, "mlfe_typer:type_modules/2 crashed"})
```

Comment:
I suggest to insert a linebreak before printing the potentially large terms. That makes it more likely to fit into constraints and will also make it easier to copy paste since it starts at column zero. Also, in case you want the whole terms to be printed, you might want to reconsider the use of `~p`.